advanced/wallets/react-wallet-v2/src/components/MultibridgeRequestModal.tsx

```diff
@@ -41,45 +47,80 @@ export default function MultibridgeRequestModal({
 
   const chainId = params?.chainId
   const request = params?.request
+  const caService = new ChainAbstractionService()
 
-  const bridge = useCallback(async () => {
-    if (!bridgingRequest) {
-      throw new Error('Bridging request is unavailable')
+  const bridgeFunds = useCallback(async (): Promise<void> => {
+    if (!bridgingTransactions) {
+      throw new Error('bridgingTransactions are unavailable')
     }
 
     const wallet = await getWallet(params)
+    console.log(
+      'Bridge funds from',
+      eip155ChainsFundsSourcedFrom,
+      'to',
+      eip155ChainFundsDestination
+    )
 
-    const asset = getAssetByContractAddress(bridgingRequest.transfer.contract)
-    if (!asset) {
-      throw new Error('Source chain asset unavailable')
-    }
-    const sourceChainAssetAddress = supportedAssets[asset][bridgingRequest.sourceChain]
-    if (!sourceChainAssetAddress) {
-      throw new Error('Source chain asset address unavailable')
+    for (const transaction of bridgingTransactions) {
+      console.log('Bridging transaction', transaction)
+      const chainId = transaction.chainId
+      const chainProvider = new providers.JsonRpcProvider(EIP155_CHAINS[chainId as TEIP155Chain].rpc)
+      const chainConnectedWallet = await wallet.connect(chainProvider)
+      const walletAddress = wallet.getAddress()
+      const gasPrice = await chainProvider.getGasPrice()
+      const gasEstimate = await chainProvider.estimateGas({
+        from: walletAddress,
+        to: transaction.to,
+        value: transaction.value,
+        data: transaction.data,
+        gasPrice: gasPrice
+      })
```

Comment:
Why do we do this? isn't this done server side?

Reply:
added the server-side value.